packages/product/src/services/product-module-service.ts

```diff
@@ -1116,22 +1116,41 @@ export default class ProductModuleService<
     return await this.productService_.softDelete(productIds, sharedContext)
   }
 
-  async restore(
+  async restore<
+    TReturnableLinkableKeys extends string = Lowercase<
+      keyof typeof LinkableKeys
+    >
+  >(
     productIds: string[],
+    {
+      returnLinkableKeys,
+    }: { returnLinkableKeys?: TReturnableLinkableKeys[] } = {
+      returnLinkableKeys: [],
+    },
```

Comment:
thought: This config param will be identical across modules, so should we extract to a type? This looks a bit messy 😄 
```suggestion
    {
      returnLinkableKeys,
    }: { returnLinkableKeys?: TReturnableLinkableKeys[] } = {
      returnLinkableKeys: [],
    },
```

Reply:
Added 2 new types for that. `SoftDeleteReturn` and `RestoreReturn`